transaction/transaction.go

```diff
@@ -20,33 +20,37 @@ const (
 //
 // Nodes is a union of the all the TxnStep.Nodes
 type Txn struct {
+	// TODO: Any good reason for this to be not just string ?
 	ID    uuid.UUID
 	Ctx   TxnCtx
 	Steps []*Step
 	Nodes []uuid.UUID
 }
 
 // NewTxn returns an initialized Txn without any steps
-func NewTxn() *Txn {
+func NewTxn(id string) *Txn {
 	t := new(Txn)
-	t.ID = uuid.NewRandom()
-	prefix := txnPrefix + t.ID.String()
 
+	if uuid.Parse(id) != nil {
+		t.ID = uuid.Parse(id)
+	} else {
+		t.ID = uuid.NewRandom()
```

Comment:
When this can happen?  If reqid is corrupted for any reason, this code silently assigns new id and continues. Parse error can be logged here to catch the same.